### ISSUE TITLE
Add react ref to table row 

### ISSUE THREAD

**Issue by mgu-certi** [ant-design/ant-design#30238]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
I'm trying to use antd-table for an infinite scroller implementation so I'm trying set an intersectionObserver to observe the last row of the table to handle a function to get more data. 

This feature would allow me to set a ref on any row use an observer to see if the ref is on the screen. I see the ref hook is already implemented in other componentes of antd, it would be helpful to have it on the table too. If it is already possible, please tell me how.

### What does the proposed API look like?
What I've been trying to do was to modify the ref tag from any rows using the onRow prop. I think that would be enough.
![Example](https://user-images.githubusercontent.com/74564423/115459230-300eb480-a1fd-11eb-8f61-ad4b1d3188e2.png)

**Comment by jameslahm** [comment]
set ref like this? [https://codesandbox.io/s/basic-usage-antd4152-forked-1c7ju?file=/index.js](https://codesandbox.io/s/basic-usage-antd4152-forked-1c7ju?file=/index.js)

**Comment by mgu-certi** [comment]
> set ref like this? https://codesandbox.io/s/basic-usage-antd4152-forked-1c7ju?file=/index.js

Thank you so much for showing me this, that's exactly what I was looking for! The second return solved the issue and I'm able to do what I was trying to... 

I really can't thank you enough for that!!

**Comment by sa1mann** [comment]
Hi @mgu-certi .
Im doing the same functionality.
I was able to get ref like this but im getting error when i do `observer.observe(this.rowRef.current)` like 

` Uncaught TypeError: Failed to execute 'observe' on 'IntersectionObserver': parameter 1 is not of type 'Element'.`
Could you please guide?

**Comment by mgu-certi** [comment]
Hello @sa1mann.

I followed [this article](https://dev.to/producthackers/intersection-observer-using-react-49ko) to implement a solution for the observer.

For the ref on the row, this is how my prop is set for the antd table component:
`
onRow={(record: any, index) => {
  if (index === rows.length - 1) {
    return {
      ref: observerRef,
    };
  }
  return {
    onClick: () => {
      // eslint-disable-next-line react/prop-types
      setSelectedRow(record);
    },
  };
}}
`

Note that rows is my dataSource (ref should exist only on the last row of the table), the const observerRef is the containerRef from the article and if I remember correctly, I had to have the second return with onClick doing something (even if it is just a console.log), else it didn't work.
